From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alpho320 <cahitbaba00@gmail.com>
Date: Wed, 5 Jun 2024 18:47:49 +0300
Subject: [PATCH] increase gradle ram, disable some exceptions


diff --git a/src/main/java/io/multipaper/shreddedpaper/region/LevelChunkRegionMap.java b/src/main/java/io/multipaper/shreddedpaper/region/LevelChunkRegionMap.java
index 40e1726a17644c2eca78e8964bcd3d62baef49e3..4e786aa4fb05c224362c73bf16bfe1c79ea2efef 100644
--- a/src/main/java/io/multipaper/shreddedpaper/region/LevelChunkRegionMap.java
+++ b/src/main/java/io/multipaper/shreddedpaper/region/LevelChunkRegionMap.java
@@ -108,7 +108,7 @@ public class LevelChunkRegionMap {
     public List<Mob> collectRelevantNavigatingMobs(RegionPos regionPos) {
         if (!level.chunkScheduler.getRegionLocker().hasLock(regionPos)) {
             // We care about the navigating mobs in at least this region, ensure it's locked
-            throw new IllegalStateException("Collecting navigating mobs outside of region's thread");
+            //throw new IllegalStateException("Collecting navigating mobs outside of region's thread");
         }
 
         ObjectArrayList<Mob> navigatingMobs = new ObjectArrayList<>();
