From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alpho320 <cahitbaba00@gmail.com>
Date: Wed, 5 Jun 2024 18:57:37 +0300
Subject: [PATCH] disable some exceptions


diff --git a/src/main/java/io/papermc/paper/util/TickThread.java b/src/main/java/io/papermc/paper/util/TickThread.java
index 699cf78c93f03298e4f48411378bbe1dfa1490d6..625d77fd1ed690bf98b3e54da55d2c5aa7d4f445 100644
--- a/src/main/java/io/papermc/paper/util/TickThread.java
+++ b/src/main/java/io/papermc/paper/util/TickThread.java
@@ -51,7 +51,7 @@ public class TickThread extends Thread {
         MinecraftServer.LOGGER.error("Extra data: " + extraData);
         MinecraftServer.LOGGER.error("Server Version: " + Bukkit.getVersion());
         MinecraftServer.LOGGER.error("Region locks currently held (regionSize=" + RegionPos.REGION_SIZE + " chunks): " + getCurrentLocksHeld());
-        MinecraftServer.LOGGER.error("Stack trace:", new Throwable());
+        //new Throwable().printStackTrace();
         //throw new IllegalStateException(reason);
     }
 
