From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alpho320 <cahitbaba00@gmail.com>
Date: Wed, 5 Jun 2024 20:36:55 +0300
Subject: [PATCH] fix CraftPlayer#teleport.


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index f9ebb2951bcee7bf7ef62de579d8eefe9afe1fb0..7219d5e7369daf3ca0909b8630230fbfe6b4b9d9 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -1496,7 +1496,8 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
             // Paper end - Teleport API
         } else {
             // The respawn reason should never be used if the passed location is non null.
-            if (!TickThread.isTickThreadFor(entity.serverLevel(), entity.position()) || !TickThread.isTickThreadFor(toWorld, CraftLocation.toBlockPosition(to))) throw new UnsupportedOperationException("Cannot respawn player synchronously"); // ShreddedPaper - make sure following will be run on this thread
+            if (!TickThread.isTickThreadFor(entity.serverLevel(), entity.position()) || !TickThread.isTickThreadFor(toWorld, CraftLocation.toBlockPosition(to)))
+                //throw new UnsupportedOperationException("Cannot respawn player synchronously"); // ShreddedPaper - make sure following will be run on this thread
             this.server.getHandle().respawnAsync(entity, toWorld, true, to, !toWorld.paperConfig().environment.disableTeleportationSuffocationCheck, null).join(); // Paper // ShreddedPaper - will be run on this thread
         }
         return true;
