From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alpho320 <cahitbaba00@gmail.com>
Date: Wed, 5 Jun 2024 21:16:14 +0300
Subject: [PATCH] some fixes


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index d89f5ac4016ee6142cda412c1b65e753b1b88f53..646321a8da482b62e01fa078159364b4af2e1c15 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -1501,7 +1501,9 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
                 Plugin plugin = SynchronousPluginExecution.getCurrentPlugin(); // ShreddedPaper - synchronous plugin execution
                 Location finalTo = to.clone();
                 entity.getBukkitEntity().getScheduler().run(plugin, task -> {
-                    this.server.getHandle().respawnAsync(entity, toWorld, true, finalTo, !toWorld.paperConfig().environment.disableTeleportationSuffocationCheck, null).join(); // Paper // ShreddedPaper - will be run on this thread
+                    this.server.getHandle().respawnAsync(entity, toWorld, true, finalTo, !toWorld.paperConfig().environment.disableTeleportationSuffocationCheck, null).thenAcceptAsync(serverPlayer -> {
+
+                    }); // Paper // ShreddedPaper - will be run on this thread
                 }, null);
             } else {
                 this.server.getHandle().respawnAsync(entity, toWorld, true, to, !toWorld.paperConfig().environment.disableTeleportationSuffocationCheck, null).join(); // Paper // ShreddedPaper - will be run on this thread
