From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alpho320 <cahitbaba00@gmail.com>
Date: Thu, 6 Jun 2024 01:44:39 +0300
Subject: [PATCH] FIX ServerChunkCache#getChunk shits.


diff --git a/src/main/java/net/minecraft/server/level/ServerChunkCache.java b/src/main/java/net/minecraft/server/level/ServerChunkCache.java
index b6cbd48f3d47a59d45398fd655dbc02f26849fbc..1a6c63cdd64f9629e3e76aa938c3af688b0c1ef9 100644
--- a/src/main/java/net/minecraft/server/level/ServerChunkCache.java
+++ b/src/main/java/net/minecraft/server/level/ServerChunkCache.java
@@ -273,7 +273,7 @@ public class ServerChunkCache extends ChunkSource {
 
         Objects.requireNonNull(completablefuture);
         if (!completablefuture.isDone()) { // Paper
-            io.papermc.paper.chunk.system.scheduling.ChunkTaskScheduler.pushChunkWait(this.level, x1, z1); // Paper - rewrite chunk system
+            io.papermc.paper.chunk.system.scheduling.ChunkTaskScheduler.pushChunkWait(this.level, x, z); // Paper - rewrite chunk system
             com.destroystokyo.paper.io.SyncLoadFinder.logSyncLoad(this.level, x, z); // Paper - Add debug for sync chunk loads
             //this.level.timings.syncChunkLoad.startTiming(); // Paper // Purpur
             chunkproviderserver_b.managedBlock(completablefuture::isDone);
